pin/pin.go

```diff
@@ -97,26 +102,38 @@ func (p *pinner) Pin(node *mdag.Node, recurse bool) error {
 			return err
 		}
 	} else {
+		if p.recursePin.HasKey(k) {
+			return errors.New("Key already pinned recursively.")
+		}
 		p.directPin.AddBlock(k)
 	}
 	return nil
 }
 
-// Unpin a given key with optional recursive unpinning
-func (p *pinner) Unpin(k util.Key, recurse bool) error {
+// Unpin a given key
+func (p *pinner) Unpin(k util.Key, recursive bool) error {
 	p.lock.Lock()
 	defer p.lock.Unlock()
-	if recurse {
-		p.recursePin.RemoveBlock(k)
-		node, err := p.dserv.Get(k)
-		if err != nil {
-			return err
+	if p.recursePin.HasKey(k) {
+		if recursive {
+			p.recursePin.RemoveBlock(k)
+			node, err := p.dserv.Get(k)
+			if err != nil {
+				return err
+			}
+
+			return p.unpinLinks(node)
+		} else {
+			return errors.New("Key pinned recursively.")
 		}
-
-		return p.unpinLinks(node)
+	} else if p.directPin.HasKey(k) {
+		p.directPin.RemoveBlock(k)
+		return nil
+	} else if p.indirPin.HasKey(k) {
+		return errors.New("Cannot unpin indirectly pinned block.")
+	} else {
+		return errors.New("Given key was not pinned.")
```

Comment:
if these are displayed to the user, the language should be consistent, how about:
```
<key> is pinned recursively
<key> is pinned indirectly. indirect pins cannot be unpinned.
<key> is not pinned
```
and actually putput the key? or, if we dont want to output the key (can go either way atm):

```
block is pinned recursively
block is pinned indirectly. indirect pins cannot be unpinned.
block is not pinned
```